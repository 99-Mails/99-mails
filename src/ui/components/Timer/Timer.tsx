import { Text } from "@chakra-ui/react";
import { FancyDate } from "@/lib/date";
import { useMemo } from "react";

type TimerProps = {
  time: number;
  isDisabled: boolean;
};

const Timer = ({ time }: TimerProps) => {
  const timer = useMemo(() => new FancyDate(), []);

  return (
    <Text data-testid="timer" as="b">
      {timer.getRemaindedMMSSFromSeconds(time)}
    </Text>
  );
};

export { Timer };
