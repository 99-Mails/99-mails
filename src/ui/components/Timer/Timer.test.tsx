import { render, screen } from "@/tests/test-utils";
import { it, expect } from "vitest";
import { Timer } from "./Timer";

it("should render correctly", () => {
  const { asFragment } = render(<Timer time={10} isDisabled={false} />);
  expect(asFragment()).toMatchSnapshot();
});

it("should show the correct time", async () => {
  render(<Timer time={10} isDisabled={false} />);

  expect(screen.getByTestId("timer")).toHaveTextContent("00:10");
});
