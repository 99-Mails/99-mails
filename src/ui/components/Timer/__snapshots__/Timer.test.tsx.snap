// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render correctly 1`] = `
<DocumentFragment>
  <b
    class="chakra-text css-0"
    data-testid="timer"
  >
    00:10
  </b>
</DocumentFragment>
`;
